mobile/lib/ui/viewer/search/result/people_section_all_page.dart

```diff
@@ -1,45 +1,91 @@
 import "dart:async";
 
+import "package:collection/collection.dart";
 import 'package:flutter/material.dart';
 import "package:flutter_animate/flutter_animate.dart";
 import "package:photos/events/event.dart";
 import "package:photos/generated/l10n.dart";
-import "package:photos/models/search/search_result.dart";
+import "package:photos/models/search/generic_search_result.dart";
+import "package:photos/models/search/hierarchical/face_filter.dart";
 import "package:photos/models/search/search_types.dart";
+import "package:photos/models/selected_people.dart";
 import "package:photos/theme/ente_theme.dart";
 import "package:photos/ui/common/loading_widget.dart";
 import "package:photos/ui/viewer/search_tab/people_section.dart";
 
-class PeopleSectionAllPage extends StatefulWidget {
+class PeopleSectionAllPage extends StatelessWidget {
   const PeopleSectionAllPage({
     super.key,
   });
 
   @override
-  State<PeopleSectionAllPage> createState() => _PeopleSectionAllPageState();
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(SectionType.face.sectionTitle(context)),
+        centerTitle: false,
+      ),
+      body: const PeopleSectionAllWidget(),
+    );
+  }
 }
 
-class _PeopleSectionAllPageState extends State<PeopleSectionAllPage> {
-  late Future<List<SearchResult>> sectionData;
+class PeopleSectionAllWidget extends StatefulWidget {
+  const PeopleSectionAllWidget({
+    super.key,
+    this.selectedPeople,
+    this.namedOnly = false,
+  });
+
+  final SelectedPeople? selectedPeople;
+  final bool namedOnly;
+
+  @override
+  State<PeopleSectionAllWidget> createState() => _PeopleSectionAllWidgetState();
+}
+
+class _PeopleSectionAllWidgetState extends State<PeopleSectionAllWidget> {
+  late Future<List<GenericSearchResult>> sectionData;
   final streamSubscriptions = <StreamSubscription>[];
 
   @override
   void initState() {
     super.initState();
-    sectionData = SectionType.face.getData(context);
+    sectionData = getResults();
 
     final streamsToListenTo = SectionType.face.viewAllUpdateEvents();
     for (Stream<Event> stream in streamsToListenTo) {
       streamSubscriptions.add(
         stream.listen((event) async {
           setState(() {
-            sectionData = SectionType.face.getData(context);
+            sectionData = getResults();
           });
         }),
       );
     }
   }
 
+  Future<List<GenericSearchResult>> getResults() async {
+    final results =
+        List<GenericSearchResult>.from(await SectionType.face.getData(context));
+
+    if (widget.namedOnly) {
+      results.removeWhere(
+        (element) =>
+            (element.hierarchicalSearchFilter as FaceFilter).personId == null,
```

Comment:
Better to use `kPersonParamID` in `params` of `elements`(`GenericSearchResult`). Let's not use anything relating to Hierarchical search here.